### PR TITLE
Update IERS_B file and adjust to new-style

### LINKED ISSUE #14380: IERS_B file download failure: 404 error
### Description

Hi,
It seems like the [URL](http://hpiers.obspm.fr/iers/eop/eopc04/eopc04_IAU2000.62-now) used to download the IERS B table is down and gives a 404 error in a new virtual environment.

### Expected behavior

I see some mention of adding fail-safe URLs in https://github.com/astropy/astropy/issues/8981, but not sure if there was a resolution for the IERS B tables?

### How to Reproduce

```python
# Put your Python code snippet here.
import astropy
from astropy.utils.iers import conf,IERS_A, IERS_A_URL, IERS_B, IERS_B_URL, IERS
from astropy.utils.data import download_file

print(astropy.__version__)
print(IERS_B_URL)
iers_b_file = download_file(IERS_B_URL, cache=True)
```

Output:
```
5.1
http://hpiers.obspm.fr/iers/eop/eopc04/eopc04_IAU2000.62-now
```
Attempting to download the IERS B table from the [given URL](http://hpiers.obspm.fr/iers/eop/eopc04/eopc04_IAU2000.62-now) gives the following error. 
`HTTPError: HTTP Error 404: Not Found`

Conversely if you attempt this in an old virtual environment (which has the file cached), I don't get this error. The analogous IERS_A_URL points to a [resource ]('https://datacenter.iers.org/data/9/finals2000A.all') that is still available.

### Versions

```
Windows-10-10.0.19045-SP0
Python 3.9.12 (tags/v3.9.12:b28265d, Mar 23 2022, 23:52:46) [MSC v.1929 64 bit (AMD64)]
astropy 5.1
Numpy 1.22.3
pyerfa 2.0.0.1
Scipy 1.8.1
Matplotlib 3.5.2
```

### PR DESCRIPTION
**Pull request by mhvk**
### Description

This pull request is to address the change in location and format of `IERS_B` data. Instructions are provided to read old-style files, but these are not automatically recognized since it should be very rare for the astropy version and the IERS_B file not to be synchronized.

EDIT: ~Fixes~ #14380 was fixed by, in the end, upstream reinstating the old IERS B tables.

### CODE REVIEW

**Comment by pllim** [comment]
Would probably need quick bugfix releases after this is merged? cc @astropy/astropy-project-release-team

**Comment by mhvk** [comment]
I guess users should not *generally* run into this, since we include the `IERS_B` file in the release, but given that there seems to be a big group in the wild which is affected, a quick bugfix release might still be called for.

File: astropy/utils/iers/tests/test_iers.py
Comment (pllim) [inline astropy/utils/iers/tests/test_iers.py:118]: I think we should open this TODO note as a proper GitHub issue and mention that issue here. Otherwise, we might forget. What do you think?

**Comment by pllim** [comment]
Failures look related. I wasn't expecting changes... did they change the format *and* the actual data?

**Comment by mhvk** [comment]
Weird failures indeed!  Checking the UT1-UTC one for `2010:001`:
```
# old data
      Date      MJD      x          y        UT1-UTC       LOD         dX        dY        x Err     y Err   UT1-UTC Err  LOD Err     dX Err       dY Err  
2010   1   1  55197   0.098695   0.192853   0.1140330   0.0005515   0.000102  -0.000071   0.000082   0.000061  0.0000304  0.0000133    0.000046    0.000049

# new data
# YR  MM  DD  HH       MJD        x(")        y(")  UT1-UTC(s)       dX(")      dY(")       xrt(")      yrt(")      LOD(s)        x Er        y Er  UT1-UTC Er      dX Er       dY Er       xrt Er      yrt Er      LOD Er
2010   1   1   0  55197.00    0.098670    0.192840   0.1141359    0.000195    0.000054   -0.001976    0.000512   0.0005004    0.000068    0.000053   0.0000598    0.000077    0.000093    0.000081    0.000118   0.0000628
```
This is not the first time the "historical" data have changed -- I think they re-analyze once in a while.

**Comment by mhvk** [comment]
I think here we would just need to update the tests, but I'd prefer not to backport that, but rather have it only appear in 5.3. So, in #14384 we just update the `IERS_B_URL` for backporting, and then for 5.3 we can do this.

**Comment by mhvk** [comment]
Converted to draft so we do not merge this before #14384 is done.

**Comment by pllim** [comment]
#14384 is merged.

**Comment by mhvk** [comment]
OK, I rebased and tried to make all the small changes that are required because the IERS B numbers have changed.

**Comment by pllim** [comment]
I don't think this is a bug fix anymore. Move change log fragment to "API change"?

**Review by pllim — APPROVED** [review]
The changes LGTM though personally I am not impacted by these changes in my work. Who should co-review?

**Comment by pllim** [comment]
RTD failure is unrelated and would be fixed by #14390

**Comment by mhvk** [comment]
Perhaps I should move the fragment to `other`? We're not changing API, really, we're just using a new calculation of IERS B (as I understand it, the main change is in the definition of the reference frame, but earlier IERS B changes have also implied changes in some of the tests).

Though the changes were substantial enough that perhaps a what's-new is warranted!? I suppose other people's precision tests may also fail.

As for other reviews, let me ping @mkbrewer, who seems to understand reference frames a lot better than I do. Specific question (following the reply from IERS quoted in https://github.com/astropy/astropy/issues/14380#issuecomment-1430178576): can we just update to the new IERS B tables, which use the 2020 IRTF (instead of 2014), or is that inconsistent with some of the other assumptions made in erfa or our code? Maybe also cc @StuartLittlefair.

**Comment by pllim** [comment]
"It looks ok," Christian Bizouard (Observatoire de Paris / SYRTE)

**Comment by pllim** [comment]
Needs a rebase with the partial revert merged. Thanks!

**Comment by mhvk** [comment]
Rebased. Also tried to rename the changelog fragment to `14382.other.rst` but pre-commit complained about that -- it can only be done in the root directory (for reasons a bit beyond me). Anyway, decided to go with "feature" since we now support the new IERS B format/analysis.

**Comment by pllim** [comment]
"other" is only acceptable if it is not inside a subpackage folder, I think.

**Comment by mkbrewer** [comment]
In case you haven't seen the memo, here it is: 
```
************************************************************************
IERS Message No. 471                                   February 01, 2023
************************************************************************


Implementation of the IERS 2020 C04 system on 14 February 2023


The IERS will be implementing the 2020 International Terrestrial
Reference Frame (ITRF2020) and making it available in its EOP products
on 14 February 2023. The official IERS 20 C04 final solution will be
available in the usual repertory:

https://hpiers.obspm.fr/iers/eop/eopc04/

(Short description:
https://hpiers.obspm.fr/iers/eop/eopc04_20/eopc04.txt)

The Rapid Service/Prediction Center solution will be available at
https://maia.usno.navy.mil/ser7.

In addition to the alignment with ITRF2020, the C04 series present a
new format given in a shortened header, and include pole rates. The
former version will be produced until June 2023 and put in the repertory
https://hpiers.obspm.fr/iers/eop/eopc04_14/. The yearly series will be
suppressed.

Users are advised to update their data from these sites on 14 February
to ensure that they are fully consistent with the new 20 C04 system. Any
data appearing on these sites before that date should be considered
preliminary.

C. Bizouard, IERS Earth Orientation Center, Paris Observatory
N. Stamatakos, IERS Rapid Service/Prediction Center,
US Naval Observatory


************************************************************************
IERS Messages are edited and distributed by the IERS Central Bureau.
If not stated otherwise, the IERS is only the distributor of the message
and is not responsible for its content.
To submit texts for distribution, please write to
<central_bureau@iers.org>.
To subscribe or unsubscribe, please create an IERS account or modify it:
https://www.iers.org/Login/Login/EN/login_node.html
or write to <central_bureau@iers.org>.
Archives: http://www.iers.org/IERS/Messages
************************************************************************

```

**Comment by mkbrewer** [comment]
This isn't IERS Bulletin B, btw. That is just preliminary monthly data. This is the final data data except for the last 30 days, which gets an updated solution twice a week. My go to site for all of this is:
https://www.iers.org/IERS/EN/DataProducts/EarthOrientationData/eop.html

**Comment by mhvk** [comment]
Thanks for posting that!  And, indeed, it isn't really IERS B, but well, at the time it seemed it was to me... (though the series that we actually download is not as "final" as you suggest or I thought; we've seen quite large parts changed, even if just by minimal amounts, of course, well within errors).

**Comment by mkbrewer** [comment]
That's interesting as the documentation says that only the last 30 days gets updated. [C04.guide.pdf](https://github.com/astropy/astropy/files/10749165/C04.guide.pdf)

**Comment by mhvk** [comment]
Yes, it is weird. See, e.g., #10376

**Comment by mhvk** [comment]
But more importantly, @mkbrewer, does it make sense to you to shift to the new format with the new IRTF, and adjust tests accordingly? I guess the e-mail you quoted suggests we may not have much choice, though I think they have moved some things back because we contacted them (and perhaps others too). Mostly wondering if somewhere else we implicitly assume some version of the IRTF.

**Comment by pllim** [comment]
Can we push this forward or does this need more discussion/changes? Thanks!

**Pull request merged**
